File: old/AutoItInterpreter/Module.cs
using System.Linq;
using System;

namespace AutoItInterpreter
{
    public static class Module
    {
        private static readonly string[] vstr;

        public static Version InterpreterVersion { get; }
        public static string GitHash { get; }


        static Module()
        {
            vstr = Properties.Resources.version.Split(new[] { '\r', '\n' }, StringSplitOptions.RemoveEmptyEntries).Select(s => s.Trim()).ToArray();

            InterpreterVersion = Version.TryParse(vstr[0], out Version v) ? v : null;
            GitHash = vstr.Length > 0 ? vstr[1] : "";

            if ((GitHash?.Length ?? 0) == 0)
                GitHash = "<unknown git commit hash>";
        }
    }
}

File: old/AutoItExpressionParser/Module.fs
namespace AutoItExpressionParser

open System.Resources
open System

module Module =
    let rec private asm = (GetModuleType <@ asm @>).Assembly

    let private lines =
        try
            let resstr = ResourceManager("AutoItExpressionParser.resources", asm).GetString("version")
            resstr.Split('\r', '\n')
            |> Array.filter (fun s -> s.Trim().Length > 0)
        with
        | _ -> [| "0.0.0.0" |]

    let Version = Version lines.[0]
    let GitHash = if lines.Length > 0 then lines.[1] else "<unknown git hash>"
